test-runner-poc/environments.js

```diff
@@ -0,0 +1,227 @@
+'use strict';
+const semver = require('semver');
+const path = require('path');
+const core = require('../lib/core');
+/**
+ * Base class for environments in projects that use the test
+ * runner
+ */
+class EnvironmentBase {
+  /**
+   * The default implementation of the environment setup
+   *
+   * @param {*} callback
+   */
+  setup(callback) {
+    callback();
+  }
+  constructor(parsedURI) {
+    this.host = parsedURI.hosts[0].host;
+    this.port = parsedURI.hosts[0].port;
+  }
+}
+const genReplsetConfig = (port, options) => {
+  return Object.assign(
+    {
+      options: {
+        bind_ip: 'localhost',
+        port: port,
+        dbpath: `${__dirname}/../db/${port}`,
+        setParameter: ['enableTestCommands=1']
+      }
+    },
+    options
+  );
+};
+function usingUnifiedTopology() {
+  return !!process.env.MONGODB_UNIFIED_TOPOLOGY;
+}
+/**
+ *
+ * @param {*} discoverResult
+ */
+class ReplicaSetEnvironment extends EnvironmentBase {
+  constructor(parsedURI, version) {
+    super(parsedURI);
+    this.setName = parsedURI.options.replicaSet || 'replset';
+    this.url = `mongodb://%s${this.host}:${this.port}/integration_tests?rs_name=rs`;
+    this.writeConcernMax = { w: 'majority', wtimeout: 30000 };
+    this.replicasetName = 'rs';
+    this.topology = function(topologyHost, topologyPort, options) {
+      topologyHost = topologyHost || this.host;
+      topologyPort = topologyPort || this.port;
+      options = Object.assign({}, options);
+      options.replicaSet = 'rs';
+      options.poolSize = 1;
+      options.autoReconnect = false;
+      if (usingUnifiedTopology()) {
+        return new core.Topology([{ topologyHost, topologyPort }], options);
+      }
+      return new core.ReplSet([{ topologyHost, topologyPort }], options);
+    };
+
+    this.nodes = generateNodesArray(parsedURI.hosts, genReplsetConfig);
```

Comment:
As Dan was saying in our meeting, we can use `replSetGetStatus)` to see what nodes are in our replicaset and we can use those nodes here. See the [docs](https://docs.mongodb.com/manual/reference/command/replSetGetStatus/#dbcmd.replSetGetStatus) for more information.